firmware/src/globals.rs

```diff
@@ -76,3 +74,24 @@ pub fn custom_exception_handler(trap_frame: &riscv_rt::TrapFrame) -> ! {
 		mtval
 	);
 }
+
+/// This will be called when a hardware exception occurs
+#[cfg(esp_xtensa)]
+#[no_mangle]
+#[link_section = ".rwtext"]
+unsafe extern "C" fn __exception(
+	cause: xtensa_lx_rt::exception::ExceptionCause,
+	context: xtensa_lx_rt::exception::Context,
+) {
+	use defmt::error;
+
+	#[cfg(esp_xtensa)]
```

Comment:
this part probably doesn't need to be feature gates since afaik *everything* xtensa is esp